### PR TITLE
[LOOP-3490] Add Health Store permissions to Tidepool Onboarding

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-3490
- Add read option to HealthKitSampleStore.authorize
- Make sampleType not optional on DoseStore

### CODE REVIEW

**Comment by darinkrauss** [comment]
LW PR for build: https://github.com/tidepool-org/LoopWorkspace/pull/561

File: LoopKit/HealthKitSampleStore.swift [hunk 1]
```diff
@@ -109,10 +109,11 @@ public class HealthKitSampleStore {
     ///
     /// - Parameters:
     ///   - toShare: Whether to request write authorization. Defaults to true.
+    ///   - read: Whether to request read authorization. Defaults to true.
```
Comment (rickpasetto) [inline LoopKit/HealthKitSampleStore.swift:112]: [nit] just the name `read` seems to indicate a verb about reading related to authorization.  Maybe consider `toRead` to match `toShare`?

**Review by rickpasetto — APPROVED** [review]
LGTM (one minor nit)

File: LoopKit/HealthKitSampleStore.swift (hunk 1 above)
Comment (darinkrauss) [reply LoopKit/HealthKitSampleStore.swift:112]: It matches a function with a similar signature on `HKHealthStore` (part of `HealthKit`).

**Pull request merged**